libraries/AP_GPS/AP_GPS_SBF.cpp

```diff
@@ -0,0 +1,368 @@
+// -*- tab-width: 4; Mode: C++; c-basic-offset: 4; indent-tabs-mode: nil -*-
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+//
+//  Septentrio GPS driver for ArduPilot.
+//  Code by Michael Oborne
+//
+
+#include "AP_GPS.h"
+#include "AP_GPS_SBF.h"
+#include <DataFlash/DataFlash.h>
+
+#if GPS_RTK_AVAILABLE
+
+extern const AP_HAL::HAL& hal;
+
+#define SBF_DEBUGGING 0
+
+#if SBF_DEBUGGING
+ # define Debug(fmt, args ...)                  \
+do {                                            \
+    hal.console->printf("%s:%d: " fmt "\n",     \
+                        __FUNCTION__, __LINE__, \
+                        ## args);               \
+    hal.scheduler->delay(1);                    \
+} while(0)
+#else
+ # define Debug(fmt, args ...)
+#endif
+
+AP_GPS_SBF::AP_GPS_SBF(AP_GPS &_gps, AP_GPS::GPS_State &_state,
+                       AP_HAL::UARTDriver *_port) :
+    AP_GPS_Backend(_gps, _state, _port)
+{	
+    sbf_msg.sbf_state = sbf_msg_parser_t::PREAMBLE1;
+}
+
+// Process all bytes available from the stream
+//
+bool
+AP_GPS_SBF::read(void)
+{
+	uint32_t now = hal.scheduler->millis();
+	
+	if (_init_blob_index < (sizeof(_initialisation_blob) / sizeof(_initialisation_blob[0]))) {

```

Comment:
the init blob is part of a define, which is in the .h file.

this is to send the other init strings once we have auto detected this gps.